File: data-service/spring-data-provider/src/main/java/org/aykhan/dataprovider/security/JwtUtils.java
package org.aykhan.dataprovider.security;

import io.jsonwebtoken.Claims;
import io.jsonwebtoken.Jwts;
import org.aykhan.dataprovider.exception.notfound.TokenNotFoundException;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.http.HttpHeaders;
import org.springframework.stereotype.Component;
import org.springframework.util.StringUtils;

import javax.servlet.http.HttpServletRequest;
import java.util.Date;

@Component
public class JwtUtils {

  private static final String BEARER = "Bearer ";
  @Value("${SECRET_KEY:secret}")
  private final String secretKey = "secret";

  public String extractToken(HttpServletRequest req) {
    String token = req.getHeader(HttpHeaders.AUTHORIZATION);
    if (StringUtils.isEmpty(token) || !token.contains(BEARER))
      throw new TokenNotFoundException();

    return token.substring(BEARER.length());
  }

  private Claims extractAllClaims(String token) {
    return Jwts
        .parser()
        .setSigningKey(secretKey)
        .parseClaimsJws(token)
        .getBody();
  }

  public String extractUsername(String token) {
    return extractAllClaims(token).getSubject();
  }

  public boolean isExpired(String token) {
    return extractAllClaims(token)
        .getExpiration()
        .before(new Date());
  }

  public boolean validateToken(String token) {
    return !isExpired(token);
  }
}

File: data-service/scala-sql-dataprovider/app/utils/JwtUtils.scala
package utils

import java.util.Date

import io.jsonwebtoken.{Claims, Jwts}
import javax.inject.Singleton

@Singleton
class JwtUtils {
  private val SECRET_KEY = "secret"
  def extractBody(token: String): Claims =
    Jwts.parser().setSigningKey(SECRET_KEY).parseClaimsJws(token).getBody

  def isExpired(token: String): Boolean =
    extractBody(token).getExpiration.before(new Date())

  def isValid(token: String): Boolean = !isExpired(token)

  def getUsername(token: String): String = extractBody(token).getSubject
}
